managed-ledger/src/test/java/org/apache/bookkeeper/mledger/impl/ManagedCursorPropertiesTest.java

```diff
@@ -232,6 +235,31 @@ void testUpdateCursorProperties() throws Exception {
         ledger.close();
 
         factory2.shutdown();
+
+        // Create a new factory to force a managed ledger close and recovery
+        ManagedLedgerFactory factory3 = new ManagedLedgerFactoryImpl(metadataStore, bkc);
+        // Reopen the managed ledger
+        ledger = factory3.open("testUpdateCursorProperties", new ManagedLedgerConfig());
+        c1 = ledger.openCursor("c1");
+
+        c1.putCursorProperty(CURSOR_INTERNAL_PROPERTY_PREFIX + "test", "test").get();
+        c1.putCursorProperty("custom4", "custom4").get();
```

Comment:
If we call the `CompletableFuture#get` method, it's better to give the test a timeout.